torch/csrc/stable/c/shim.h

```diff
@@ -37,6 +37,29 @@ AOTI_TORCH_EXPORT AOTITorchError torch_library_impl(
     void (*fn)(StableIValue*, uint64_t, uint64_t),
     uint64_t extension_build_version);
 
+struct StableListOpaque;
+using StableListHandle = StableListOpaque*;
+
+// returns an owning reference of a StableList. callee is responsible for
+// freeing memory.
+AOTI_TORCH_EXPORT AOTITorchError
+torch_new_list_reserve_size(size_t size, StableListHandle* ret);
```

Comment:
interestingly, this does not correspond to any constructor of std::vector, though I like avoiding a second crossing of the ABI boundary just to do a reserve.